### PR TITLE
Automated cherry pick of #66832: Detect if GCE PD udev link is wrong and try to correct it

### PR DESCRIPTION
**Pull request by msau42**
Cherry pick of #66832 on release-1.9.

#66832: Detect if GCE PD udev link is wrong and try to correct it

#66925: Rework multi-volume test to use StatefulSet

### CODE REVIEW

**Review by saad-ali — APPROVED** [review]
/lgtm
/approve

**Comment by davidz627** [comment]
/assign @mbohlool 
for cherry-pick approval

**Comment by msau42** [comment]
/hold
e2e test is failing in multi-zone environment, I am working on a fix

**Comment by msau42** [comment]
/hold cancel
I cherry picked the e2e fix too

**Comment by msau42** [comment]
I had to make extra changes to the e2e test because the StatefulSet utils were using v1beta1 in 1.9

**Comment by davidz627** [comment]
@msau42 I think there are failing tests

**Comment by msau42** [comment]
yes i'm debugging.  the e2e test utils are different in 1.9 because statefulset was beta.

**Comment by msau42** [comment]
the new e2e looks good now

File: test/e2e/storage/persistent_volumes.go [hunk 1]
```diff
@@ -300,4 +302,122 @@ var _ = SIGDescribe("PersistentVolumes", func() {
...
+				Expect(err).NotTo(HaveOccurred())
+				ssTester.WaitForRunningAndReady(1, ss)
+
+				By("Deleting the StatefulSet but not the volumes")
+				// Scale down to 0 first so that the Delete is quick
+				ss, err = ssTester.Scale(ss, 0)
+				Expect(err).NotTo(HaveOccurred())
+				ssTester.WaitForStatusReplicas(ss, 0)
+				// Set OrphanDependent=false so it's deleted synchronously
+				err = c.AppsV1beta1().StatefulSets(ns).Delete(ss.Name, &metav1.DeleteOptions{OrphanDependents: new(bool)})
```
Comment (davidz627) [inline test/e2e/storage/persistent_volumes.go]: is `new(bool)==false? `

**Review by davidz627 — COMMENTED** [review]
1 question, lgtm besides that

File: test/e2e/storage/persistent_volumes.go (hunk 1 above)
Comment (msau42) [reply test/e2e/storage/persistent_volumes.go]: yup

**Comment by davidz627** [comment]
/assign @mbohlool 
for cherry-pick approval

**Comment by davidz627** [comment]
ping @mbohlool

**Pull request merged**